tests/test_pdf_uploader.py

```diff
@@ -0,0 +1,95 @@
+import unittest
+from unittest.mock import patch, MagicMock
+from dsst_etl.upload_pdfs import PDFUploader
+from dsst_etl.db import get_db_session
+from dsst_etl.models import Documents, Provenance, Works
+from pathlib import Path
+
+class TestPDFUploader(unittest.TestCase):
+
+    @patch('dsst_etl.upload_pdfs.boto3.client')
+    def setUp(self, mock_boto_client):
+        # Mock S3 client
+        self.mock_s3_client = MagicMock()
+        mock_boto_client.return_value = self.mock_s3_client
+
+        # Start a new database session and transaction
+        self.session = get_db_session()
+        self.session.begin_nested()  # Start a nested transaction
+
+        # Initialize PDFUploader with the session
+        self.uploader = PDFUploader(self.session)
+
+    def tearDown(self):
```

Comment:
After running this test, I still have records left in my Works, Documents, and Provenance tables. Please check this so the test does not leave records.

Reply:
This appears fixed in the latest version. Thanks!